cairo-libs/packages/cometbft/src/types.cairo

```diff
@@ -314,6 +317,34 @@ pub struct PublicKey {
     pub sum: Sum,
 }
 
+#[generate_trait]
+pub impl PublicKeyImpl of PublicKeyTrait {
+    fn verify(self: @PublicKey, msg: Span<u8>, signature: Span<u8>) {
+        match self.sum {
+            Sum::Ed25519(pk) => {
+                assert(signature.len() == 64, CometErrors::INVALID_SIGNATURE_LENGTH);
+                let r_sign = signature
+                    .slice(0, 32)
+                    .try_into()
+                    .unwrap(); // Never fails as length is 32.
+                let s_sign = signature
+                    .slice(32, 32)
+                    .try_into()
+                    .unwrap(); // Never fails as length is 32.
+
+                assert(pk.len() == 32, CometErrors::INVALID_PUBKEY_LENGTH);
```

Comment:
shouldn't this `assert` be on the top of the block? avoid redundant gas usage?

```cairo
assert(signature.len() == 64, CometErrors::INVALID_SIGNATURE_LENGTH);
assert(pk.len() == 32, CometErrors::INVALID_PUBKEY_LENGTH);

let r_sign = ...
```

Reply:
[88f2bf3](https://github.com/informalsystems/ibc-starknet/pull/361/commits/88f2bf3c087fd1f697ecf4e671cdbc5652d4ef5f)